opencog/nlp/learn/make-disjuncts.scm

```diff
@@ -134,14 +121,29 @@
 		(if (equal? word firstword)
 			(if (null? nodes)
 				'()
-				(cons (string-append linkname "+") (extract-disjunct word (cdr nodes))))
+				(cons (cons secondword (string-append linkname "+")) (get-related-words-disjuncts word (cdr nodes))))
 			(if (equal? word secondword)
 				(if (null? nodes)
 					'()
-					(cons (string-append linkname "-") (extract-disjunct word (cdr nodes))))
+					(cons (cons firstword (string-append linkname "-")) (get-related-words-disjuncts word (cdr nodes))))
 				(if (null? nodes)
 					'()
-					(cons '() (extract-disjunct word (cdr nodes)))))))
+					(get-related-words-disjuncts word (cdr nodes))))))
+		'()))
+
+(define (arrange-disjuncts-in-order listofdisjuncts sentence)
```

Comment:
Why do you need to arrange disjuncts in order ?? I'm confused.  

Also: You seem to have two concepts in your code, one of which you call a disjunct, but it isn't really.   A disjunct is this: (MB+ MC-)  but your are using "word-connector-lists"  ((the . MB+) (played. MC-))  It would be less confusing for me if you called this second thing "WCL" in the code, and explained that a wcl looks like ...  ((the . MB+) (played. MC-))